components/director/internal/domain/runtime/resolver.go

```diff
@@ -189,11 +200,22 @@ func (r *Resolver) Runtime(ctx context.Context, id string) (*graphql.Runtime, er
 func (r *Resolver) RegisterRuntime(ctx context.Context, in graphql.RuntimeInput) (*graphql.Runtime, error) {
 	convertedIn := r.converter.InputFromGraphQL(in)
 
+	if err := r.selfRegManager.PrepareRuntimeForSelfRegistration(ctx, &in); err != nil {
+		return nil, err
+	}
```

Comment:
I guess this block is not in the transaction because in case of failure we won't know if we need to clean up the self reg(delete call for the cloning) in the defer block below, right? We will need to check the type of error?

Reply:
Well, that is one thing but I did not put that in the transaction mostly because it looks like a bad idea to keep an open DB connection while calling external services.